x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/index_view_logic.ts

```diff
@@ -130,10 +130,6 @@ export const IndexViewLogic = kea<MakeLogicType<IndexViewValues, IndexViewAction
     ],
   },
   events: ({ actions }) => ({
-    afterMount: () => {
-      const { indexName } = IndexNameLogic.values;
-      actions.startFetchIndexPoll(indexName);
-    },
```

Comment:
`IndexViewLogic` (and `IndexNameLogic`) are both mounted in the `SearchIndexRouter` which also contains the `useEffect` call to update the name when it changes based on the URL. There are no other calls to `IndexViewLogic.mount()`.

https://github.com/elastic/kibana/blob/602b6d645a9668a6c48a5478fd27ff248bb78a32/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/search_index_router.tsx#L33-L45

Reply:
Nice!

FYI logics are auto-mounted when called, so you don't need explicit calls to `.mount()` to trigger `afterMount`, but this will still work because the router wraps all components where IndexViewLogic is called from, so this mount should always get called first.